miio/yeelight_dual_switch.py

```diff
@@ -0,0 +1,254 @@
+            "switch_1_default_state=%s, "
+            "switch_1_off_delay=%s, "
+            "switch_2_state=%s, "
+            "switch_2_default_state=%s, "
+            "switch_2_off_delay=%s, "
+            "interlock=%s, "
+            "flex_mode=%s, "
+            "rc_list=%s>"
+            % (
+                self.switch_1_state,
+                self.switch_1_default_state,
+                self.switch_1_off_delay,
+                self.switch_2_state,
+                self.switch_2_default_state,
+                self.switch_2_off_delay,
+                self.interlock,
+                self.flex_mode,
+                self.rc_list,
+            )
+        )
+        return s
+
+
+class YeelightDualControlModule(MiotDevice):
+    """Main class representing the air humidifier which uses MIoT protocol."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "First Switch Status: {result.switch_1_state}\n"
+            "First Switch Default State: {result.switch_1_default_state}\n"
+            "First Switch Delay: {result.switch_1_off_delay}\n"
+            "Second Switch Status: {result.switch_2_state}\n"
+            "Second Switch Default State: {result.switch_2_default_state}\n"
+            "Second Switch Delay: {result.switch_2_off_delay}\n"
+            "Interlock: {result.interlock}\n"
+            "Flex Mode: {result.flex_mode}\n"
+            "RC list: {result.rc_list}\n",
+        )
+    )
+    def status(self) -> DualControlModuleStatus:
+        """Retrieve properties"""
+        p = [
+            "switch_1_state",
+            "switch_1_default_state",
+            "switch_1_off_delay",
+            "switch_2_state",
+            "switch_2_default_state",
+            "switch_2_off_delay",
+            "interlock",
+            "flex_mode",
+            "rc_list",
+        ]
+        properties = [
+            {"did": k, **v}
+            for k, v in filter(lambda item: item[0] in p, _MAPPING.items())
+        ]
+        values = self.get_properties(properties)
+        return DualControlModuleStatus(
+            dict(map(lambda v: (v["did"], v["value"]), values))
+        )
+
+    @command(
+        click.argument("switch", type=EnumType(Switch)),
+        default_output=format_output("Turn On {switch} Switch"),
+    )
+    def on(self, switch: Switch):
+        """Turn On Switch."""
+        if switch == Switch.First:
+            return self.set_property("switch_1_state", True)
+        elif switch == Switch.Second:
+            return self.set_property("switch_2_state", True)
+
+    @command(
+        click.argument("switch", type=EnumType(Switch)),
+        default_output=format_output("Turn Off {switch} Switch"),
```

Comment:
```suggestion
        default_output=format_output("Turn off switch {switch}"),
```
Is likely better ordering for console output?